.github/workflows/validate.yml

```
@@ -0,0 +1,81 @@
+# Runs cargo check, test, fmt and clippy on every push for default members.
+# Based on https://github.com/actions-rs/meta/blob/master/recipes/quickstart.md
+
+on: [push]
+
+name: Validate
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install native dependencies
+        run: sudo apt-get update && sudo apt-get install libasound2-dev
+
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Run cargo check
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+
+  test:
+    name: Run Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install native dependencies
+        run: sudo apt-get update && sudo apt-get install libasound2-dev
+
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Run cargo test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+  lints:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install native dependencies
+        run: sudo apt-get update && sudo apt-get install libasound2-dev
+
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          components: rustfmt, clippy
+
+      - name: Run cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          ## fmt doesn't compile the crates so we can include dtrace-sys with --all
+          args: --all --check
+
+      - name: Run cargo clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
```

Elsewhere we also have:
```suggestion
          command: clippy
          args: --all-targets -- -D warnings
```
I think `--all-targets` make this check also tests, examples etc, and `-D warnings` makes this actually fail the CI if any warnings are found. But I'd be curious if these 2 are still needed, be encouraged to double-check.

---

You're right. Done in e65d6b5c20e5d35331fbe43771b815c8c7500dcd where I also added `--all-features` which does what you'd expect.